tests/subsys/net/lib/nrf_cloud/src/main.c

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <unity.h>
+#include <zephyr.h>
+#include <stdio.h>
+#include <net/nrf_cloud.h>
+#include <mock_nrf_cloud_transport.h>
+#include "nrf_cloud_fsm.h"
+
+/* This is required since unity_main return int and
+ * zephyr expects main not to return any value
+ */
+extern int unity_main(void);
+
+/* Simple event handler needed for the init function to
+ * have a success initialization in the test below
+ */
+void event_handler(const struct nrf_cloud_evt *const evt)
+{
+	switch (evt->type) {
+	case NRF_CLOUD_EVT_READY:
+		printk("Device is connected and ready");
+		break;
+	case NRF_CLOUD_EVT_ERROR:
+		printk("Failed to connect to cloud");
+		break;
+	default:
+		break;
+	}
+}
+
+/* This function runs before each test */
+void setUp(void)
+{
+	/* Uninit to change the internal state to idle */
+	__wrap_nct_uninit_Ignore();
+	nrf_cloud_uninit();
+}
+
+/* This function runs after each completed test */
+void tearDown(void)
+{
+	/* Uninit to reset the internal state to idle */
+	__wrap_nct_uninit_Expect();
+	nrf_cloud_uninit();
+}
+
+/* The library can only be initialized when the internal state is
+ * idle, so the test verifies that nrf_cloud_init function will
+ * return with permission denied error (-EACCES) when not idle
+ */
+void test_init_not_idle(void)
+{
+	struct nrf_cloud_init_param params = {
+		.event_handler = event_handler,
+		.client_id = NULL,
+	};
+
+	TEST_ASSERT_EQUAL_MESSAGE(STATE_IDLE, nfsm_get_current_state(),
+		"internal state should be idle");
+
+	/* Call nrf_cloud_init twice with the first one
+	 * being successful and check if the second one
+	 * will fail since the state will not be idle
+	 */
+	__wrap_nct_init_IgnoreAndReturn(0);
+	nrf_cloud_init(&params);
+
+	int ret = nrf_cloud_init(&params);
+
+	TEST_ASSERT_NOT_EQUAL_MESSAGE(STATE_IDLE, nfsm_get_current_state(),
+		"internal state should not be idle");
```

Comment:
Perhaps rephrase this as:
"nrf_cloud lib should not be in the idle state after initialization"